### ISSUE TITLE
Tracking issue for UI redesign

### ISSUE THREAD

**Issue by ishitatsuyuki** [keepassxreboot/keepassxc#1443]
The current UI is poorly designed and looks cluttered. This issue acts as a thread for collecting ideas.

- Refresh the icons first
  * KP2Android has a quite good set
  * materialdesignicons is also a good candidate
- Refactor the UI
  * Rewrite to QML?
    - QtQuick 2 is not native looking and requires Qt 5.7
  * Or incrementally update the widgets code
  * Using dialogs instead of "tabs"

**Comment by moqmar** [comment]
https://icons8.com/ is a great iconset too (with even multiple styles) and already comes with all the icons for biger websites. They're also [free for open source projects](https://icons8.com/articles/weve-contributed-an-icon-to-imageoptim/)

The UI is actually not *bad* in my opinion, it's just ugly. So, improving it based on the current codebase is probably fine for now.

I think getting rid of the borders between panes (like the Windows Explorer does), using dialogs as you suggested, cleaner icons for the menu bar (ideally the ones provided by the system), starting with QT_QPA_PLATFORMTHEME=gtk2 on GTK distributions (no idea how to detect that though), and some more tweaks would already help a lot.

The question is anyway: should this be an incremental change towards a better, but familiar UI, or a full redesign with a completely different concept? The latter would probably make it easier to choose a different framework (maybe even GTK), but it'd also be more work.

**Comment by t-aus-m** [comment]
If I were to choose, I would definetly go for the full redesign. Not because the current one is _really_ bad, but rather because the UX - as you mentioned - could use some improvements aswell. 
Just mentioning a few points here, that need to be done (imo):
- A better implementation of multiple URLs per Entry
- A way to create custom entries more easily

This list could shurely be extended, I just wanted to state my point.

**Comment by TheZ3ro** [comment]
Like I said in https://github.com/keepassxreboot/keepassxc/issues/1564#issuecomment-369755981
Redesigning the UI will cause mass rage of people that explicitly want the KeePass-like UX

Anyway just a note about the multiple URLs per entry. We can't do much since the KDBX format allow 1 URL. We can indeed do some magic workaround but it's totally UI unrelated, needs new core implementation

**Comment by phoerious** [comment]
Technically, the KeePass2 XML format allows for arbitrarily many URLs. They will just show up as advanced string fields in other applications.

**Comment by TheZ3ro** [comment]
The XML format allow only 1 URL since there can be only 1 String element with `Key="URL"`.
Using custom field is part of the magic workaround I was talking above, that will need a core implementation other than UI, so my point really was: out of the scope of this issue

**Comment by phoerious** [comment]
Yes, but URL is also just a convention. We can as well introduce URL2, URL3, etc.

**Comment by moqmar** [comment]
But what if you can fix thousands of user's workflows, and only break a few. KeePassXC, as it is right now, is the only open-source offline password manager with a great set of features. Now, is the project's goal to achieve that all the users of the last two abandoned similar projects are happy, or to get more people to actually use a) a password manager and b) open source software? It's pretty hard to do both at once, but if it's just the first one, I'll happily create another fork of this project, because I think we need to get open source software, and especially password managers available (and accessible) for everyone.

TL;DR: if this project should have a future, in my opinion it can't get by without a massive UI overhaul.

Multiple URL fields is also an issue that needs to be addressed by the UI, no matter how it works in the core.

**Comment by TheZ3ro** [comment]
You are right, open source. It means that we are not paid for what we do, and a "massive UI overhaul" isn't a thing that we can develop in a small amount of time. In that very same time we can fix many other feature so we need to choose where to invest our time.

I'm not saying "no redesign", I'm asking if the design is really a priority or there are other must-have feature worth it.

I would really like KeePassXC to have the same userbase as 1Password or LastPass but this is not going to happen (probably even without the UI redesign).

**Comment by moqmar** [comment]
I know that, that's also why I said a progression to a more modern UI would be a good solution for now. I can also contribute some of my time for this project as I really like the concept behind KeePass(XC), maybe that way we can achieve a full redesign in the long run.

**Comment by ishitatsuyuki** [comment]
On IRC, I've heard that some people prefer the OS-native look and feel, which QtQuick2 doesn't provide. Do you think this is an actual blocker?

**Comment by phoerious** [comment]
QtQuick2 is very specialized for mobile design these days. I know that KDE is using it for the desktop and it works well there, but they also put a lot of work into it to create their own platform look.
I think we should redesign conservatively. Enhance things progressively. With a full redesign I fear that we will break a lot of things. We will break a lot of users' workflows and we will also break a lot of actual features, because a redesign is complex and will introduce bugs. We don't really have the manpower to support a complete redesign. I'm not against KeePassXC 2.85 looking entirely different from KeePassXC 2.3, but I'm concerned that we cannot deliver quality if we do a full one-off redesign.

**Comment by droidmonkey** [comment]
I am closing this in favor of the new UI Redesign Epic
